### PR TITLE
Type Mapping Refactor

### PR DESCRIPTION
**Pull request by DnOberon**
## Description
This is a complete refactor of the Type Mapping system of Deep Lynx. 

Type mappings now have children records called Transformations. Those transformations are what are combined with an import's payloads to create nodes and edges for insert. New storage layers for both were created.

Completely redid the UI for handling Type Mappings and Transformations. This touches the DataMapping and DataImports views, with their subsequent pieces.

**NOTE**: This PR completely breaks the data processing loop of Deep Lynx. All data will fail to ingest, even if a type mapping with valid transformations is created. The next planned feature/update to Deep Lynx will re-enable data processing.

## How Has This Been Tested?
Tested the UI
Test suite passes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/data_storage/import/data_staging_storage.ts [hunk 1]
```diff
@@ -63,9 +58,20 @@ export default class DataStagingStorage extends PostgresStorage {
     }
 
     public async List(importID: string, offset:number, limit:number): Promise<Result<DataStagingT[]>>{
+        if(limit === -1) { // allow return all with a negative number
```
Comment (jerenmb) [inline src/data_storage/import/data_staging_storage.ts:61]: Is this behavior that we have used anywhere else or is it unique? If it's unique, are there other patterns we have followed elsewhere?

File: src/data_storage/import/data_staging_storage.ts [hunk 2]
```diff
@@ -63,9 +58,20 @@ export default class DataStagingStorage extends PostgresStorage {
...
+            return super.rows<DataStagingT>(DataStagingStorage.listAllStatement(importID))
+        }
         return super.rows<DataStagingT>(DataStagingStorage.listStatement(importID, offset, limit))
     }
 
+    public async ListAndSort(importID: string, offset:number, limit:number, sortBy:string, sortDesc: boolean): Promise<Result<DataStagingT[]>>{
+        if(sortDesc) {
```
Comment (jerenmb) [inline src/data_storage/import/data_staging_storage.ts:68]: Would it be easier/simpler to have one listSortStatement and pass sortDesc through to it? Either is fine with me though.

File: src/data_storage/import/type_transformation_storage.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
+import Result from "../../result"
+import PostgresStorage from "../postgresStorage";
+import {QueryConfig} from "pg";
+import PostgresAdapter from "../adapters/postgres/postgres";
+import {dataSourceT, DataSourceT} from "../../types/import/dataSourceT";
+import {typeTransformationT, TypeTransformationT} from "../../types/import/typeMappingT";
+
+/*
+* ImportAdapterStorage encompasses all logic dealing with the manipulation of the Import Adapter
```
Comment (jerenmb) [inline src/data_storage/import/type_transformation_storage.ts:9]: Comment should be updated

File: src/data_storage/import/type_transformation_storage.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+* class in a data storage layer.
+*/
+export default class TypeTransformationStorage extends PostgresStorage{
+    public static tableName = "data_type_mapping_transformations";
```
Comment (jerenmb) [inline src/data_storage/import/type_transformation_storage.ts:13]: Perhaps an overly verbose table name?

File: src/data_storage/import/data_staging_storage.ts (hunk 1 above)
Comment (DnOberon) [reply src/data_storage/import/data_staging_storage.ts:61]: it's being used somewhere else now. I should update the rest of the storage layers with it.

File: src/data_storage/import/data_staging_storage.ts (hunk 2 above)
Comment (DnOberon) [reply src/data_storage/import/data_staging_storage.ts:68]: I could do that, yeah

File: src/data_storage/import/type_transformation_storage.ts (hunk 4 above)
Comment (DnOberon) [reply src/data_storage/import/type_transformation_storage.ts:13]: going to keep the table name as its descriptive and shows the relationship between mappings and transformations. Also helps in the join statements

**Review by jerenmb — APPROVED** [review]

**Pull request merged**